src/app/users/getManageConsoleRoles.js

```diff
@@ -0,0 +1,69 @@
+const config = require('../../infrastructure/config');
+const { sendResult } = require('../../infrastructure/utils');
+const { getUserDetails} = require('./utils');
+const { getServiceById } = require('../../infrastructure/applications')
+const { listRolesOfService, getSingleUserService, getSingleInvitationService, updateUserService } = require('../../infrastructure/access');
+
+const getSingleServiceForUser = async (userId, organisationId, serviceId, correlationId) => {
+  const userService = userId.startsWith('inv-') ? await getSingleInvitationService(userId.substr(4), serviceId, organisationId, correlationId) : await getSingleUserService(userId, serviceId, organisationId, correlationId);
+  const application = await getServiceById(serviceId, correlationId);
+
+  return {
+    id: userService.serviceId,
+    roles: userService.roles,
+    name: application.name
+  }
+};
+
+const addOrChangeManageConsoleServiceTitle = (userManageRoles, manageConsoleRoleIds) => {
+  let userManageConsoleRoleIds = [];
+  let result = false;
+  for(let i=0; i < userManageRoles.roles.length; i++) {
+    userManageConsoleRoleIds.push(userManageRoles.roles[i].id);
+  }
+
+  for(let i=0; i < manageConsoleRoleIds.length; i++) {
+   if (userManageConsoleRoleIds.includes(manageConsoleRoleIds[i])) {
+    result = true;
+    break;
+  }
+}
+return result;
+}
+
+const checkIfRolesChanged = (rolesSelectedBeforeSession, newRolesSelected) => {
+  return !(JSON.stringify(rolesSelectedBeforeSession.sort()) === JSON.stringify(newRolesSelected.sort()));
+}
+
+const getManageConsoleRoles = async (req, res) => {
+  
+  const manage = await getServiceById('manage');

```

Comment:
I'm quite happy for this additional call to be here, as the performance hit will be negligible, but if we're configuring the DfE org ID, adding this to the config as well may be useful 👍 If not now then down the line.

Reply:
Thanks Jared, the app now adds the manage console service if the user doesn't have it, and will only update if they do have the service.